Lawnicons as icon pack (core changes)

naveenccmsd opened the pull request:
## Description

- LawnIcons as icon pack
- Two apk default one (icons are dark ) and another apk with light icons(white lines)
- Introduce automation to reduce some maintance work. 
      - Since lawncions will act as icon pack, we need to add components details to appfilter.xml , drawabale.xml and grayscale_icon_map.xml .  Even now developers are creating svg  files and vector assests.
      - Idea is to let developer create svg file and appfilter.xml file alone and helper script to create others file during build.

Demo:

https://user-images.githubusercontent.com/7315975/201498198-fad3c058-89b6-4510-802b-1f560ee8bd76.mp4


Issues:
I see svgs and vector asset is not in sync since they added or created manually. They both need to be in sync inorder for lawnicons to work.

Action Items:

- [x] Update configs to recognise lawicons as icon pack
- [x] Update workflow to build 2 apk 
- [x] Create helper class to create xml and vector automatically
- [x] create/update svg as per vector asset  (noticed some svgs width and height updated in vector asset ). 
- [x] Update Activity for Some apps [app/assets/appfilter_notMapped.xml](https://github.com/LawnchairLauncher/lawnicons/pull/815/files#diff-78d8683a60dc7b78b04e8837085b82c04e2ebf1d5725958b4e6b9bd4d99fedb0)

Previous Icon mapping : **1098**
Current Icon Mapping : **2472**
Pending to update App Activty : **79** (Added activty to all apps avalible in playstore except paid ones)

Goooler commented on gradle.properties:
```diff
@@ -2,7 +2,7 @@
 android.defaults.buildfeatures.buildconfig=false
 android.defaults.buildfeatures.renderscript=false
-android.defaults.buildfeatures.resvalues=false
+android.defaults.buildfeatures.resvalues=true
```
Enable this flag in app's build.gradle.kts

Goooler commented on settings.gradle.kts:
```diff
@@ -35,6 +35,4 @@ dependencyResolutionManagement {
     }
 }
 
-rootProject.name = "Lawnicons"
```
Why need to remove rootProject's name?

Goooler commented on .github/workflows/build_debug_apk.yml:
```diff
@@ -48,12 +48,14 @@ jobs:
           fi
       - name: Build debug APK
-        run: ./gradlew app:assembleDebug
+        run: ./gradlew app:assembleDarkDebug
+      - name: Build debug Light APK
+        run: ./gradlew app:assembleLightDebug
```
`app:assembleDebug` should be enough, will build 2 apks.

Goooler commented:
Can we convert other module to a Gradle module so that get rid of maven?

naveenccmsd commented:
@Goooler I don't have handson on gradle , how about converting when this if fully working ?

Goooler commented:
Alright.

Goooler commented on build.gradle.kts:
```diff
@@ -10,6 +10,7 @@ plugins {
 
 allprojects {
     apply(plugin = "org.jmailen.kotlinter")
+    apply(plugin = "kotlin-kapt")
```
Shouldn't use kapt here, see #715

Goooler replied:
15c650830dab97abd8ed467459734c26a9e9d1f1

Goooler commented on app/build.gradle.kts:
```diff
@@ -111,6 +131,7 @@ dependencies {
     implementation("io.github.fornewid:material-motion-compose-core:0.10.3")
     implementation("com.google.dagger:hilt-android:$hiltVersion")
     annotationProcessor("com.google.dagger:hilt-android-compiler:$hiltVersion")
+    annotationProcessor ("com.google.dagger:hilt-compiler:$hiltVersion")
```
Why need to apply `hilt-compiler` here?

naveenccmsd replied:
The apk didn't launch , encountered classnot found exception , I tried many way nothing worked.

It was working after adding kapt compiler.
kapt ("com.google.dagger:hilt-compiler:$hiltVersion")

naveenccmsd replied:
@Goooler I see you changed to annotationProcessor ("com.google.dagger:hilt-compiler:$hiltVersion")

But getting below exception now while launching appp
```


  Process: app.lawnchair.lawnicons, PID: 30361
                                                                                                    java.lang.RuntimeException: Unable to instantiate application app.lawnchair.lawnicons.LawniconsApplication package app.lawnchair.lawnicons: java.lang.ClassNotFoundException: Didn't find class "app.lawnchair.lawnicons.LawniconsApplication" on path: DexPathList[[zip file "/data/app/~~o-8lXcw91hImyuOEZLyQHA==/app.lawnchair.lawnicons-uObhy24i-P4ZEGnIwQesoQ==/base.apk"],nativeLibraryDirectories=[/data/app/~~o-8lXcw91hImyuOEZLyQHA==/app.lawnchair.lawnicons-uObhy24i-P4ZEGnIwQesoQ==/lib/arm64, /system/lib64, /system/system_ext/lib64]]
                                                                                                    	at android.app.LoadedApk.makeApplication(LoadedApk.java:1465)
                                                                                                    	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7540)
                                                                                                    	at android.app.ActivityThread.access$1600(ActivityThread.java:315)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2286)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:226)
                                                                                                    	at android.os.Looper.loop(Looper.java:313)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8751)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
                                                                                                    Caused by: java.lang.ClassNotFoundException: Didn't find class "app.lawnchair.lawnicons.LawniconsApplication" on path: DexPathList[[zip file "/data/app/~~o-8lXcw91hImyuOEZLyQHA==/app.lawnchair.lawnicons-uObhy24i-P4ZEGnIwQesoQ==/base.apk"],nativeLibraryDirectories=[/data/app/~~o-8lXcw91hImyuOEZLyQHA==/app.lawnchair.lawnicons-uObhy24i-P4ZEGnIwQesoQ==/lib/arm64, /system/lib64, /system/system_ext/lib64]]
                                                                                                    	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:259)
                                                                                                    	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
                                                                                                    	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
                                                                                                    	at android.app.AppComponentFactory.instantiateApplication(AppComponentFactory.java:76)
                                                                                                    	at androidx.core.app.CoreComponentFactory.instantiateApplication(CoreComponentFactory.java:52)
                                                                                                    	at android.app.Instrumentation.newApplication(Instrumentation.java:1177)
                                                                                                    	at android.app.LoadedApk.makeApplication(LoadedApk.java:1457)
                                                                                                    	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7540) 
                                                                                                    	at android.app.ActivityThread.access$1600(ActivityThread.java:315) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2286) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:226) 
                                                                                                    	at android.os.Looper.loop(Looper.java:313) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8751) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135) 
                                                                                                    	Suppressed: java.lang.NoClassDefFoundError: Failed resolution of: Lapp/lawnchair/lawnicons/Hilt_LawniconsApplication;
                                                                                                    		at java.lang.VMClassLoader.findLoadedClass(Native Method)
                                                                                                    		at java.lang.ClassLoader.findLoadedClass(ClassLoader.java:738)
                                                                                                    		at java.lang.ClassLoader.loadClass(ClassLoader.java:363)
                                                                                                    		... 15 more
```

Goooler replied:
Need to adopt Napt.

naveenccmsd replied:
Found that this is related to new Napt version 1.18 which happend on two days before. If I revert to one 1.17 it's workig fine

Goooler replied:
Could you file an issue to Napt?

Goooler replied:
https://github.com/sergei-lapin/napt/issues/16

Goooler replied:
Solved in #830

Goooler commented:
Please convert `others` module's Java to Kotlin, and fix the build.

naveenccmsd commented:
> Can we convert other module to a Gradle module so that get rid of maven?

Gradle conversion done

SuperDragonXD commented:
Just a note: Once you have finished all your changes, you should update `CONTRIBUTING.md` to show how to create new drawable links to `appfilter.xml`.

Goooler commented:
Please convert `others` module's Java to Kotlin, let's keep this project as a pure Kotlin project.

naveenccmsd commented:
> 



> Just a note: Once you have finished all your changes, you should update `CONTRIBUTING.md` to show how to create new drawable links to `appfilter.xml`.

Updated   `CONTRIBUTING.md`

naveenccmsd commented:
> Please convert `others` module's Java to Kotlin, let's keep this project as a pure Kotlin project.

Converted to kotlin

naveenccmsd commented on app/src/main/kotlin/app/lawnchair/lawnicons/util/GetIconInfo.kt:
```diff
@@ -9,7 +9,7 @@ fun Context.getIconInfo(): List<IconInfo> {
     try {
-        val xmlId = R.xml.grayscale_icon_map
+        val xmlId = R.xml.theme_config
         if (xmlId != 0) {
```
we don't need grayscale_icon_map for lawnchair to provide theme icons but to provide backward compaitable with older lawnicons we should keep that in lawnchair.

So used new name _theme_config_.

Goooler commented on others/src/main/kotlin/app/lawnchair/lawnicons/helper/CommonUtil.kt:
```diff
@@ -0,0 +1,71 @@
+        return rootElement.elements(path)
+    }
+
+    fun sortedMapByValues(map: Map<String, String>): Map<String, String>? {
+        val result2: MutableMap<String, String> = LinkedHashMap()
+        map.entries.stream().sorted(java.util.Map.Entry.comparingByValue())
+            .forEachOrdered { (key, value): Map.Entry<String, String> ->
+                result2[key] = value
+            }
+        return result2
+    }
```
Could you please convert these 3 files to be more idiomatic, do not just apply auto-convert results here.

naveenccmsd replied:
I'm not getting you , Did you mean write clean code in kotlin insted of auto-convert from java to kotlin?

Goooler replied:
First you use auto-convert, secondly you should make them more idiomatic and clean.

naveenccmsd replied:
Refactored as much I can.

Goooler commented:
Please test b807b2f8fb94a2ea25f9bcab468ee355ceb5e7db.

naveenccmsd commented:
> Please test [b807b2f](https://github.com/LawnchairLauncher/lawnicons/commit/b807b2f8fb94a2ea25f9bcab468ee355ceb5e7db).

It's working fine.

Goooler commented on app/src/main/AndroidManifest.xml:
```diff
@@ -17,6 +17,36 @@
                 <action android:name="android.intent.action.MAIN" />
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
+            <!-- Lawnchair Launcher -->
+            <intent-filter>
+                <action android:name="app.lawnchair.icons.THEMED_ICON" />
+                <category android:name="android.intent.category.DEFAULT" />
+            </intent-filter>
+            <!-- ADW Launcher -->
+            <intent-filter>
+                <action android:name="org.adw.launcher.THEMES" />
+                <category android:name="android.intent.category.DEFAULT" />
+            </intent-filter>
+            <!-- ADW Launcher Custom Icon Picker -->
+            <intent-filter>
+                <action android:name="org.adw.launcher.icons.ACTION_PICK_ICON" />
+                <category android:name="android.intent.category.DEFAULT" />
+            </intent-filter>
+            <!-- Apex Launcher -->
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+                <category android:name="com.anddoes.launcher.THEME" />
+            </intent-filter>
+            <!-- GO Launcher -->
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+                <action android:name="com.gau.go.launcherex.theme" />
+                <category android:name="android.intent.category.DEFAULT" />
+            </intent-filter>
+            <!-- Atom Launcher -->
+            <intent-filter>
+                <action android:name="com.dlto.atom.launcher.THEME" />
+            </intent-filter>
```
If we can support Nova launcher here?

Goooler commented on app/build.gradle.kts:
```diff
@@ -18,8 +18,8 @@ android {
         targetSdk = 31
-        versionCode = 3
-        versionName = "1.2.0"
+        versionCode = 4
+        versionName = "1.3.0"
```
Version change should not be included in this PR

naveenccmsd replied:
@Goooler I agree , The reason I added here is , there was some conflict supporting both older version of lawnicons and new version which is this iconpack.

So added a condition in below PR to used old method to apply theme icons untill versioncode 3.

https://github.com/LawnchairLauncher/lawnchair/pull/3116/files#diff-0f635825b22aff4c4eb71e3540b8efd6a93670c6158aa5381177267236e72c11R61

Or atleast the version code should be updated to fully work.

daywalk3r666 commented on CONTRIBUTING.md:
```diff
@@ -57,3 +57,11 @@ Here’s how to add an icon to&nbsp;Lawnicons:
   Here, the part before the '/' character i.e _org.telegram.messenger_ is the package name [PACKAGE_NAME] and the part after that i.e _org.telegram.messenger_.DefaultIcon is the Activity name [APP_ACIVITY_NAME].
 
+### Using 3rd Party Apps.
+  1. Download [Numix Icon Request] (https://f-droid.org/is/packages/org.numixproject.iconsubmit/). 
```
i would use this app for icon requests
https://github.com/Kaiserdragon2/IconRequest/releases

naveenccmsd replied:
@daywalk3r666 yes it's better let me update it,

daywalk3r666 commented:
Idk how to help you with the appfilter_notmapped but here some activities

<-- Camera --> 
	`<item component="ComponentInfo{org.lineageos.aperture/org.lineageos.aperture.CameraLauncher}" drawable="camera"/>`
<-- Feeder -->
	`<item component="ComponentInfo{com.nononsenseapps.feeder.play/com.nononsenseapps.feeder.ui.MainActivity}" drawable="feeder"/>`
<-- Next DB Navigator -->
	`<item component="ComponentInfo{com.deutschebahn.app.android/db.vendo.android.vendigator.view.main.MainActivity}" drawable="next_db_navigator"/>`
<-- dynamic​Spot -->
`	<item component="ComponentInfo{com.jamworks.dynamicspot/com.jamworks.dynamicspot.SettingsHome}" drawable="dynamicspot"/>`
	<!-- SmartPack -->
	`<item component="ComponentInfo{com.smartpack.kernelmanager/com.smartpack.kernelmanager.activities.StartActivity}" drawable="smartpack"/>
`

naveenccmsd commented:
@daywalk3r666 , wow thank you , will add them

daywalk3r666 commented:
<-- Threema Libre -->
	`<item component="ComponentInfo{ch.threema.app.libre/ch.threema.app.activities.MainActivity}" drawable="threema_libre"/>`
<-- Threema huawei -->
	`<item component="ComponentInfo{ch.threema.app.hws/ch.threema.app.activities.MainActivity}" drawable="threema_huawei"/>`
<-- Threema F-Droid -->
	`<item component="ComponentInfo{ch.threema.app.fdroid/ch.threema.app.activities.MainActivity}" drawable="threema_fdroid"/>`

But i think threema fdroid is canceled. And org.lineage.selfie too.

daywalk3r666 commented on app/assets/appfilter.xml:
```diff
@@ -1,26 +1,21 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
 <resources>
-
```
I test the same strings. It won't worked for me. I must add this `item
        component=":CALENDAR"
        drawable="calendar" />`

naveenccmsd replied:
Lawnchair didn't designed to support this. 
<item component=":CALENDAR" drawable="calendar" />
Can you check if the activity is correctly mentioned ? Latest dev installed ?Tried  retsrat lawnchir ?

It was working fine samsung calendar even without themed

daywalk3r666 replied:
I can test it on the last ci build. I test it on kvaesitso

naveenccmsd replied:
Downloaded kvaesitso , dynamic icons applied to Samsung calendar.  Please check you calendar package and activity added here.


![WhatsApp Image 2022-11-17 at 11 17 20 PM](https://user-images.githubusercontent.com/7315975/202519889-de1ca150-a21d-46f7-ad17-afbf7013d321.jpeg)

daywalk3r666 reviewed (commented):
<-- Pizza Hut Deutschland -->
	`<item component="ComponentInfo{com.pizzahut.germany/com.pizzahut.germany.MainActivity}" drawable="pizza_hut_deutschland"/>`

naveenccmsd commented:
> <-- Pizza Hut Deutschland --> `<item component="ComponentInfo{com.pizzahut.germany/com.pizzahut.germany.MainActivity}" drawable="pizza_hut_deutschland"/>`

@daywalk3r666  I have added Activity for all the icons expect , paid apps or not avaliable in playstore or no supported for my coyntry.


I have remove appfilter_non_mapped and moved them  to greyscale mapping file so atleaset it would to applied to older style.

Chefski reviewed (commented):
`watchbuddy.svg` is using fill instead of stroke

daywalk3r666 commented:
<-- Netto-App -->
	`<item component="ComponentInfo{com.valuephone.vpnetto/de.netto.b2c.activity.FragmentActivity}" drawable="nettoapp"/>`

naveenccmsd commented:
> `watchbuddy.svg` is using fill instead of stroke

Fixed it

naveenccmsd commented:
@Chefski If possible could you please get the activity details for all new icons untill this pr merged.

Below is new contribution guide 
https://github.com/naveenccmsd/lawnicons/blob/Iconpack/CONTRIBUTING.md
 
Or they can use below app to get the details.
https://github.com/Kaiserdragon2/IconRequest/releases

daywalk3r666 reviewed (commented):
<-- Netto-App -->
	`<item component="ComponentInfo{com.valuephone.vpnetto/de.netto.b2c.activity.FragmentActivity}" drawable="nettoapp"/>`
<-- Joyn -->
	`<item component="ComponentInfo{de.prosiebensat1digital.seventv/de.prosiebensat1digital.playerpluggable.testapp.splash.SplashActivity}" drawable="joyn"/>`

<-- ntv -->
	`<item component="ComponentInfo{de.lineas.lit.ntv.android/de.lineas.ntv.tablet.MainActivity}" drawable="ntv"/>`

daywalk3r666 commented:
Yes svg is good. I only push the wrong xml

naveenccmsd commented:
@Goooler any chance to merging this pr?

SuperDragonXD reviewed (changes requested):
I'd recommend squashing some commits in this PR so it's much more cleaner when we'll rebase this (instead of a squash merge).

Also, how about you change the `others` module to a better name, like `xmlgen` or `svgprocessor` (probably too long)? It would make the module name self-descriptive.

Lastly (optional), I'd recommend making a PR in Lawnchair on exposing `show_component_names`. Can be useful so you don't need to use ADB.

Goooler commented on app/build.gradle.kts:
```diff
@@ -48,6 +48,25 @@ android {
+            dimension = "variant"
+            applicationIdSuffix = ".light"
+            versionNameSuffix = "-light"
+            resValue("string", "apps_name", "Lawnicons-light")
```
Agree. We can merge "Add others module" first.

naveenccmsd replied:
@Goooler & @SuperDragonXD , I will squash this commits once I login from computer..

I was thinking rename light/dark to black/white varient, any suggestions? Or should we keep it light/dark itself

naveenccmsd commented:
> I'd recommend squashing some commits in this PR so it's much more cleaner when we'll rebase this (instead of a squash merge).
> 
> Also, how about you change the `others` module to a better name, like `xmlgen` or `svgprocessor` (probably too long)? It would make the module name self-descriptive.
> 
> Lastly (optional), I'd recommend making a PR in Lawnchair on exposing `show_component_names`. Can be useful so you don't need to use ADB.

I have removed the commits related to drawable cleanup to have this PR more cleaner and created separate PR https://github.com/LawnchairLauncher/lawnicons/pull/884 which can be merged once this PR is merged.

Goooler commented:
First, you should separate a PR with changes in `others` module.

naveenccmsd commented:
@Goooler created separate pr for othersmodule (svgprocessor) https://github.com/LawnchairLauncher/lawnicons/pull/885

SuperDragonXD commented:
Please update the description of your PR to reflect new changes.

Also, I think you should update `CONTRIBUTING.md` to show that you'll need to add the code too in `grayscale_icon_map.xml`.

naveenccmsd commented:
> 

The description doesn't change because the functionally is still the same , just the submodule is merged in separate or but configuration of those those functionality actually covered in this pr.

grayscale_icon_map.xml is not required to update further, it has some app entries where I can't able find the mapping because it's not available in Play Store or is paid version. I have added those to grayscale_icon_map.xml so it can applied to older version of lawnchair. Every entry from grayscale_icon_map.xml has to be moved to appfilter.xml at sometime in future.

SuperDragonXD commented:
> grayscale_icon_map.xml is not required to update further, it has some app entries where I can't able find the mapping because it's not available in Play Store or is paid version. I have added those to grayscale_icon_map.xml so it can applied to older version of lawnchair. Every entry from grayscale_icon_map.xml has to be moved to appfilter.xml at sometime in future.

So ig you're stating that some (sooner: all) icons will only be available in LC12.1 Alpha 5 (or the later dev builds)? If so it isn't really maintaining backwards compatibility.

naveenccmsd commented:
> > grayscale_icon_map.xml is not required to update further, it has some app entries where I can't able find the mapping because it's not available in Play Store or is paid version. I have added those to grayscale_icon_map.xml so it can applied to older version of lawnchair. Every entry from grayscale_icon_map.xml has to be moved to appfilter.xml at sometime in future.
> 
> So ig you're stating that some (sooner: all) icons will only be available in LC12.1 Alpha 5 (or the later dev builds)? If so it isn't really maintaining backwards compatibility.

Actuallty the future versions ie 1.3.0 and above will work on both new and old version of lawnchair(all lawnchair versions).

Only issue I see now is newer version of Lawnchair will identify themed icons based on intent insted of hardcoded package name , so it required new lawnicons(themed intent).


Ps, If people installing new lawnchair version , they might need to update lanwicons as well. but it can fixed with a change in lawnchair.

naveenccmsd commented:
> > > grayscale_icon_map.xml is not required to update further, it has some app entries where I can't able find the mapping because it's not available in Play Store or is paid version. I have added those to grayscale_icon_map.xml so it can applied to older version of lawnchair. Every entry from grayscale_icon_map.xml has to be moved to appfilter.xml at sometime in future.
> > 
> > 
> > So ig you're stating that some (sooner: all) icons will only be available in LC12.1 Alpha 5 (or the later dev builds)? If so it isn't really maintaining backwards compatibility.
> 
> Actuallty the future versions ie 1.3.0 and above will work on both new and old version of lawnchair(all lawnchair versions).
> 
> Only issue I see now is newer version of Lawnchair will identify themed icons based on intent insted of hardcoded package name , so it required new lawnicons(themed intent).
> 
> Ps, If people installing new lawnchair version , they might need to update lanwicons as well. but it can fixed with a change in lawnchair.

This will be fixed with PR https://github.com/LawnchairLauncher/lawnchair/pull/3154

After this all the icons will be supported with older version of lawnchair.

But on new lawnchair version , the icons present in _grayscale_icon_map.xml_ will not be applied because it will get appiled based on applfilter, So those entry need to moved to appfilter (future).

SuperDragonXD commented:
I know that Lawnicons 1.3 still works on Lawnchair 12 Alpha 5 and above. What I meant to say was that the icons (new icon links) added in `appfilter.xml` won't be shown on older Lawnchair builds (that uses `grayscale_icon_map.xml`). It won't be an issue for now, but when 200+ icons are on `appfilter` and not in `grayscale`, it *will* be an issue of not maintaining backwards compatibility.

----

Also I have checked on the UI for applying Lawnicons for themed icons *in Lawnchair*, and it really needs a rework:

```
How to enable it:
1. Use Lawnicons or any supported icon pack.
2. Enable Themed Icons

Issues with it:
* You can enable Themed Icons without selecting a proper icon pack, and it won't work. This WILL cause many support issues on how to enable Themed Icons.
* There is no indicator on what icon packs are supported by Themed Icons

UI Rework:
  Home Screen preferences:
  > Icon Style (go to Icon Style)
  > Themed Icons (go to Themed Icons) 

  Themed Icons preferences:
  > Themed Icons (disabled, show in home screen, show in both)
  > List of icon packs that support Themed Icons (if none, default to Lawnicons and don't show)
  > Transparent Themed Icons (switch)

  "Select Custom Icon" UI
  > <list of normal icon packs>
  > <list of icon packs with Themed Icons support>

(Btw, do this in a new PR in Lawnchair)
```

Also, update README.md to show that Lawnicons can now be used as an icon pack.

Lastly, update some files (e.g. `GetIconInfo.kt`) to use `appfilter` instead of `grayscale_icon_map`.

naveenccmsd commented:
@SuperDragonXD actually grayscale_icon_map.xml is still available at new versions as well, only difference is they will get generated in runtime. Whatever icons present in appfilter will still work in same way. 

Only impact is 45 entries in grayscale_icon_map.xml will only work with older lawnchair version.
They need to be moved to appfilter.xml

SuperDragonXD commented:
Now that the other PR has been merged, I could merge this PR now.

Maybe fix the merge conflicts though first.

Goooler commented on app/build.gradle.kts:
```diff
@@ -61,6 +61,25 @@ android {
         }
     }
 
+    flavorDimensions += "variant"
+    productFlavors {
+        create("dark") {
+            dimension = "variant"
+            resValue("string", "apps_name", "Lawnicons")
+        }
+        create("light") {
+            dimension = "variant"
+            applicationIdSuffix = ".light"
+            versionNameSuffix = "-light"
+            resValue("string", "apps_name", "Lawnicons (light)")
+        }
+    }
```
```suggestion
    flavorDimensions += "product"
    productFlavors {
        create("dark") {
            dimension = "product"
            resValue("string", "apps_name", "Lawnicons")
        }
        create("light") {
            dimension = "product"
            applicationIdSuffix = ".light"
            versionNameSuffix = "-light"
            resValue("string", "apps_name", "Lawnicons (light)")
        }
    }
```

Goooler replied:
Don't use `"variant"` here, it's simular to `variant`.

Goooler commented on .github/workflows/build_release_apk.yml:
```diff
@@ -69,5 +71,5 @@ jobs:
           TELEGRAM_CI_BOT_TOKEN: ${{ secrets.TELEGRAM_CI_BOT_TOKEN }}
           TELEGRAM_CI_CHANNEL_ID: ${{ secrets.TELEGRAM_CI_CHANNEL_ID }}
           TELEGRAM_TEAM_GROUP_ID: ${{ secrets.TELEGRAM_TEAM_GROUP_ID }}
-          ARTIFACT_DIRECTORY: artifacts/release-apk
+          ARTIFACT_DIRECTORY: artifacts/release-apk/dark/release
```
Why it's `artifacts/release-apk/dark/release`?

naveenccmsd replied:
I have added this to ship dark(default) version alone to telegram channel. Let mw know If it's required to upload other varient to telegram channel as well. I will check and update it.

Goooler replied:
You should upload all flavors?

Goooler commented on app/build.gradle.kts:
```diff
@@ -61,6 +61,25 @@ android {
+            resValue("string", "apps_name", "Lawnicons (light)")
+        }
+    }
+    sourceSets.named("dark") {
+        res.srcDirs("src/dark/res/")
+    }
+    sourceSets.named("light") {
+        res.srcDirs("src/light/res/")
+    }
```
```suggestion
```
Unnecessary, they are the default paths.

naveenccmsd replied:
Is it so? I was not aware of it. Thanks for suggestion

SuperDragonXD commented on CONTRIBUTING.md:
````diff
@@ -17,24 +17,53 @@ In addition to the above, the icons must have an outlined (not filled) style. If
 
     ```xml
-    <icon drawable="@drawable/files_by_google" package="com.google.android.apps.nbu.files" name="Files by Google" />
+    <item component="ComponentInfo{app.lawnchair.lawnicons/app.lawnchair.lawnicons.MainActivity}" drawable="lawnicons"/> 
````
I'd recommend keeping the same app/package name as before (Files by Google) for consistency with the rest of the steps/images.

This also includes the bottom section.

SuperDragonXD commented on CONTRIBUTING.md:
```diff
@@ -17,24 +17,53 @@ In addition to the above, the icons must have an outlined (not filled) style. If
-
 1. Done! You’re ready to open a pull request. Please set `develop` as the base&nbsp;branch.
+
+## How to find App Package and App Activity of your Android App
```
```suggestion
## How to find the package and activity name of your app
```

SuperDragonXD commented on CONTRIBUTING.md:
```diff
@@ -17,24 +17,53 @@ In addition to the above, the icons must have an outlined (not filled) style. If
+## How to find App Package and App Activity of your Android App
+
+
+### Using adb via computer/phone with OTG.
```
The periods are unnesessary in the headers.

SuperDragonXD commented on CONTRIBUTING.md:
```diff
@@ -17,24 +17,53 @@ In addition to the above, the icons must have an outlined (not filled) style. If
+
+### Using adb via computer/phone with OTG.
+
+  1. Connect your Android device or emulator to the PC/Mac/Phone via otg and open the App whose details you want to inspect i.e Whatsapp.
+  1. Open a Command Prompt or Terminal window and use ‘adb devices‘
+  1. Finally, type the below-given command to get the information about the currently open application.
```
```suggestion
  1. Connect your Android device or emulator to the PC/Mac/Phone via otg and open the app whose details you want to inspect, i.e. Files by Google.
  1. Open a new Command Prompt or Terminal window and input `adb devices`
  1. Finally, type the below-given command to get the information about the currently open application.
```

SuperDragonXD commented on CONTRIBUTING.md:
```diff
@@ -17,24 +17,53 @@ In addition to the above, the icons must have an outlined (not filled) style. If
+  1. Open a Command Prompt or Terminal window and use ‘adb devices‘
+  1. Finally, type the below-given command to get the information about the currently open application.
+
+  #### For Mac/Linux:
```
I think bolding this text instead of making it a subheader makes more sense:

```suggestion
  **For Mac/Linux**:
```

SuperDragonXD commented on CONTRIBUTING.md:
````diff
@@ -17,24 +17,53 @@ In addition to the above, the icons must have an outlined (not filled) style. If
+  ```
+  ![](./contributing-image-5.png)
+
+  Here, the part before the '/' character i.e _org.telegram.messenger_ is the package name [PACKAGE_NAME] and the part after that i.e _org.telegram.messenger_.DefaultIcon is the Activity name [APP_ACIVITY_NAME].
````
```suggestion
  Here, the part before the `/` character i.e `org.telegram.messenger` is the package name `[PACKAGE_NAME]` and the part after that i.e `org.telegram.messenger_.DefaultIcon` is the Activity name `[APP_ACIVITY_NAME]`.
```

SuperDragonXD reviewed (changes requested):
Just requesting a few minor changes to CONTRIBUTING.md

SuperDragonXD commented on CONTRIBUTING.md:
````diff
@@ -17,24 +17,53 @@ In addition to the above, the icons must have an outlined (not filled) style. If
+  adb shell dumpsys window | grep -E 'mCurrentFocus'  
+  ```
+
+  #### For Windows:
````
```suggestion
  **For Windows**:
```
